### PR TITLE
Only snapshot deterministic h3dex keys, precalc on h3dex load

### PR DESCRIPTION
**Pull request by Vagabond**
This should fix the current snapshot agreement issues.

### CODE REVIEW

**Comment by madninja** [comment]
how the heck did you figure out it was only those two keys!

**Comment by Vagabond** [comment]
> how the heck did you figure out it was only those two keys!

a patched `snapshot diff`

**Comment by Vagabond** [comment]
To elaborate, the `h3dex cache` was calculated at a point in time depending on when a validator upgraded, thus there was some drift. Some hexes, depending on when the validator upgraded, would or would not have a cache associated and the cache key would not be re-calculated until there was "activity" on the hex.

So the change in this PR is to exclude those non-deterministic keys from the snapshot, and force them to be recalculated on snapshot load.

**Comment by Vagabond** [comment]
Also, I tested this with a drifted light gateway validator and was able to reconcile the drift with the new snapshots (there's still drift when the non-deterministic keys are not excluded).

File: src/ledger/v1/blockchain_ledger_v1.erl [hunk 1]
```diff
@@ -6473,7 +6484,10 @@ snapshot_h3dex(Ledger) ->
     case config(?poc_targeting_version, Ledger) of
         {ok, N} when N >= 6 ->
             H3CF = h3dex_cf(Ledger),
-            snapshot_raw(H3CF, Ledger);
+            snapshot_raw(H3CF, Ledger, fun({<<"count-", _/binary>>, _}) -> false;
+                                           ({<<"gws-", _/binary>>, _}) -> false;
```
Comment (xandkar) [inline src/ledger/v1/blockchain_ledger_v1.erl:6488]: Would be good to leave a comment for posterity - why these two.

File: src/ledger/v1/blockchain_ledger_v1.erl [hunk 2]
```diff
@@ -6217,9 +6217,20 @@ load_pocs(_, PoCs, Ledger) ->
...
+
+snapshot_raw(CF, L, Filter) ->
     %% XXX Preserve order to ensure that snapshots hash deterministically.
     %% Since rocks folds are lexicographic - we can just reverse:
-    lists:reverse(cache_fold(L, CF, fun({_, _}=KV, KVs) -> [KV | KVs] end, [])).
+    lists:reverse(cache_fold(L, CF, fun({_, _}=KV, KVs) ->
+                                            case Filter(KV) of
+                                                true ->
+                                                    [KV | KVs];
+                                            false ->
```
Comment (xandkar) [inline src/ledger/v1/blockchain_ledger_v1.erl:6230]:
```suggestion
                                                false ->
```

**Review by jadeallenx — APPROVED** [review]

**Review by andymck — APPROVED** [review]

**Pull request merged**